lib/targets/cuda/target_cuda.cmake

```diff
@@ -198,12 +198,44 @@ set(QUDA_MULTIGRID_MRHS_LIST ${QUDA_MULTIGRID_MRHS_DEFAULT_LIST} CACHE STRING "T
 mark_as_advanced(QUDA_MULTIGRID_MRHS_LIST)
 message(STATUS "QUDA_MULTIGRID_MRHS_LIST=${QUDA_MULTIGRID_MRHS_LIST}")
 
+option(QUDA_ENABLE_MMA "Enabling using tensor core" ON)
+mark_as_advanced(QUDA_ENABLE_MMA)
+
 if(QUDA_MULTIGRID)
-  option(QUDA_ENABLE_MMA "Enabling using tensor core" ON)
-  mark_as_advanced(QUDA_ENABLE_MMA)
 
-  option(QUDA_MULTIGRID_SETUP_USE_SMMA "Enabling using SMMA (3xTF32/3xBF16/3xFP16) for multigrid setup" ON)
-  mark_as_advanced(QUDA_MULTIGRID_SETUP_USE_SMMA)
+  set(QUDA_MULTIGRID_MMA_TYPES "SIMT" "SMMA" "1xFP16" "3xFP16" "1xTF32" "3xTF32" "3xBF16" "DEFAULT")
```

Comment:
Can you check at `CMake` configure time that a valid MMA type is being used.  At the moment it will let you select `TF32` types and try to build even if you are on sm_70.  Best to catch this kind of error at configure time.